### PR TITLE
Add option to specify initial browser session token

### PR DESCRIPTION
**Pull request by kira-bruneau**
This will allow us to manage browser session tokens our self for the public app oauth flow.

fixes ggt-1457

### CODE REVIEW

File: packages/api-client-core/src/GadgetConnection.ts [hunk 1]
```diff
@@ -114,13 +115,21 @@ export class GadgetConnection {
 
   enableSessionMode(options?: true | BrowserSessionAuthenticationModeOptions) {
     this.authenticationMode = AuthenticationMode.BrowserSession;
+
+    let sessionTokenStore;
```
Comment (jasong689) [inline packages/api-client-core/src/GadgetConnection.ts:119]: nit: Could we type this as `BrowserStorage` like in the class property? Doesn't need to be if it's inferred but I don't think it is

File: packages/api-client-core/src/GadgetConnection.ts [hunk 2]
```diff
@@ -114,13 +115,21 @@ export class GadgetConnection {
...
+      sessionTokenStore = window.localStorage;
     } else if (options.storageType == BrowserSessionStorageType.Session) {
-      this.sessionTokenStore = window.sessionStorage;
+      sessionTokenStore = window.sessionStorage;
     } else {
-      this.sessionTokenStore = new InMemoryStorage();
+      sessionTokenStore = new InMemoryStorage();
+    }
+
+    if (options && isObject(options) && "initialToken" in options && options.initialToken) {
```
Comment (jasong689) [inline packages/api-client-core/src/GadgetConnection.ts:128]: nit: I don't think the check of `options` is necessary here. I believe `isObject` returns false if `options` is undefined

**Review by jasong689 — APPROVED** [review]
Code looks good, but are there any tests we can add/should add for this?

**Comment by jasong689** [comment]
Might need to find out from @kristianpd or @airhorns if we need to bump the package version

File: packages/api-client-core/src/GadgetConnection.ts (hunk 2 above)
Comment (kira-bruneau) [reply packages/api-client-core/src/GadgetConnection.ts:128]: Oh yep, thanks!

File: packages/api-client-core/src/GadgetConnection.ts (hunk 1 above)
Comment (kira-bruneau) [reply packages/api-client-core/src/GadgetConnection.ts:119]: we should need to explicitly type this here, it should be inferred from use.

**Comment by kira-bruneau** [comment]
> Might need to find out from @kristianpd or @airhorns if we need to bump the package version

Oh yep, I think we'll just need to bump the minor version.

File: packages/api-client-core/src/ClientOptions.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ export interface ClientOptions {
 
 /** Options to configure a specific browser-based authentication mode */
 export interface BrowserSessionAuthenticationModeOptions {
+  initialToken?: string;
```
Comment (airhorns) [inline packages/api-client-core/src/ClientOptions.ts:13]: Can you add a doc comment to explain what this does / if anyone else should use it?

**Review by airhorns — APPROVED** [review]
LGTM!

**Pull request merged**